dsindex.go

```diff
@@ -51,7 +56,46 @@ func (ei entityIndex) String() string {
 
 type DatastoreIndex map[string][]entityIndex
 
-func LoadIndex(data []byte) (DatastoreIndex, error) {
+type datastoreAdminClient struct {
+	client *admin.DatastoreAdminClient
+	context context.Context
+}
+
+func NewDatastoreAdminClient(ctx context.Context) datastoreAdminClient{
+	c, err := admin.NewDatastoreAdminClient(ctx)
+	if err != nil {
+		panic(fmt.Sprintf("Error creating DatastoreAdminClient: %s", err))
+	}
+
+	dac := datastoreAdminClient{
+		client: c,
+		context: ctx,
+	}
+	return dac
+}
+
+func (c datastoreAdminClient) GetIndexDatastore(project string, readyOnly bool) (DatastoreIndex, error) {
+	req := &dsadmin.ListIndexesRequest{
+		ProjectId: project,
+	}
+
+	it := c.client.ListIndexes(c.context, req)
+	for {
```

Comment:
not following why we are iterating over the indexes here, then passing the full result set to LoadIndexDatastore. The list index request struct has a filter field. Would it make sense to build a filter of what we want, then in the iterator below, transform to our DatastoreIndex; instead of LoadIndexDatastore?